### PR TITLE
Updates and Windows Support

### PR DESCRIPTION
**Pull request by IT-MikeS**
Made a few changes and fixes. Nice work on this app, hope my pull helps you!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,8 @@
 node_modules
-package-lock.json
```
Comment (leerob) [inline .gitignore]: It's recommended to include the `package-lock`, correct? I must have missed that.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+### Disclaimer
+This tool should only be used on videos you have express permission to download in any form, such as your own content.
```
Comment (leerob) [inline README.md]: Could this be moved to the bottom of the README?

Also, small wording nits:
> This program should only be used on non-copyrighted material.

I think that should suffice.

File: app/containers/app.container.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ class AppContainer extends Component {
     };
 
     this.interval = null;
-    this.api_key = '5a142a55461d5fef016acfb927fee0bd';
+    this.api_key = 'yt-mp3.com';
```
Comment (leerob) [inline app/containers/app.container.js]: Hmm, why was this changed? So the working URL is now `http://www.yt-mp3.com/fetch?v=${id}&apikey=yt-mp3.com`?

File: main.js [hunk 4]
```diff
@@ -1,71 +1,83 @@
-'use strict';
```
Comment (leerob) [inline main.js]: Should `'use strict'` have been removed?

File: main.js [hunk 5]
```diff
@@ -1,71 +1,83 @@
...
-const app = electron.app;
-const BrowserWindow = electron.BrowserWindow;
-const Menu = electron.Menu;
-const ipcMain = electron.ipcMain;
+const {app, BrowserWindow, ipcMain, Menu} = require('electron');
+const isDevMode = require('electron-is-dev');
 const path = require('path');
 
-require('electron-reload')(__dirname + '/public');
+if (isDevMode)
```
Comment (leerob) [inline main.js]: Could you add brackets to this if statement?

File: main.js [hunk 6]
```diff
@@ -1,71 +1,83 @@
...
 let mainWindow;
 
-function createWindow() {
-  const browserOptions = {
-    width: 800,
-    height: 600,
-    maximizeable: false,
-    icon: path.join(__dirname, '/public/img/logo.png')
-  }
+function createWindow(){
```
Comment (leerob) [inline main.js]: Did you reformat this file using tabs or something?

File: main.js [hunk 7]
```diff
@@ -1,71 +1,83 @@
...
+        icon: path.join(__dirname, '/public/img/logo.png')
+    };
+
+    mainWindow = new BrowserWindow(browserOptions);
+    mainWindow.loadURL('file://' + __dirname + '/index.html');
 
-  mainWindow = new BrowserWindow(browserOptions);
-  mainWindow.loadURL('file://' + __dirname + '/index.html');
+    // Open the DevTools.
+    // mainWindow.webContents.openDevTools();
```
Comment (leerob) [inline main.js]: This can probably be removed now since you added the menu option. Good idea btw 👍

File: main.js [hunk 8]
```diff
@@ -1,71 +1,83 @@
...
-      { label: "Undo", accelerator: "CmdOrCtrl+Z", selector: "undo:" },
-      { label: "Redo", accelerator: "Shift+CmdOrCtrl+Z", selector: "redo:" },
-      { type: "separator" },
-      { label: "Cut", accelerator: "CmdOrCtrl+X", selector: "cut:" },
-      { label: "Copy", accelerator: "CmdOrCtrl+C", selector: "copy:" },
-      { label: "Paste", accelerator: "CmdOrCtrl+V", selector: "paste:" }
-    ]
-  }
-  ];
+    //if developing add dev menu option to menu bar
```
Comment (leerob) [inline main.js]: Nit: Prefer to capitalize the first letter of inline comments and have a space between // and the first word, e.g.
```
// If developing add dev menu option to menu bar
```

File: package.json [hunk 9]
```diff
@@ -28,13 +29,14 @@
     "babel-preset-es2015": "^6.6.0",
     "babel-preset-react": "^6.5.0",
     "babelify": "^7.2.0",
-    "electron-reload": "^0.2.0",
-    "electron-prebuilt": "^0.36.0",
```
Comment (leerob) [inline package.json]: I didn't realize this project is now retired. Makes sense to replace it.

**Comment by leerob** [comment]
Thanks for submitting a PR! This definitely helps me 😄

File: .gitignore (hunk 1 above)
Comment (IT-MikeS) [reply .gitignore]: Yeah providing package-lock is recommended now.

File: README.md (hunk 2 above)
Comment (IT-MikeS) [reply README.md]: I think that would be fine, I'm not sure on it either way, but i doubt it hurts to have it somewhere

File: app/containers/app.container.js (hunk 3 above)
Comment (IT-MikeS) [reply app/containers/app.container.js]: correct, I tried with the key you had and got invalid api key errors so I dug into their site and found them using that one so I tried it and it worked.

File: main.js (hunk 4 above)
Comment (IT-MikeS) [reply main.js]: Its not required in for the V8 engine and can cause issue with newer es6+ syntax

File: main.js (hunk 5 above)
Comment (IT-MikeS) [reply main.js]: you could but it's generally accepted if you only have one line following an if to just use the intent syntax.

File: main.js (hunk 6 above)
Comment (IT-MikeS) [reply main.js]: Im not sure what happened here

File: main.js (hunk 7 above)
Comment (IT-MikeS) [reply main.js]: yep no need for those comments

File: main.js (hunk 8 above)
Comment (IT-MikeS) [reply main.js]: sure, if you have a comment convention i'm all for it, mine is just simply that they exist

File: package.json (hunk 9 above)
Comment (IT-MikeS) [reply package.json]: Yeah, also it being in dependencies not devDeps will add bloat to the final package.

**Comment by IT-MikeS** [comment]
Corrections made in accordance with PR review

**Comment by IT-MikeS** [comment]
Made some UI changes I think make if more apparent what is happening, also  leverages the response from yt-mp3 for progress status, so if it takes a while the user can see real conversion progress.

File: app/containers/app.container.js (hunk 3 above)
Comment (leerob) [reply app/containers/app.container.js]: Awesome, nice work!

File: app/containers/app.container.js [hunk 10]
```diff
@@ -5,13 +5,13 @@ import ProgressBar from '../components/ProgressBar';
...
 
-
 class AppContainer extends Component {
   constructor(props) {
     super(props);
     this.state = {
       showProgressBar: false,
-      progress: 0
+      progress: 0,
+        progressMessage: 'Sending request..',
```
Comment (leerob) [inline app/containers/app.container.js]: I think there's a funny indent here. `progress` and `progressMessage` should be indented the same.

File: app/containers/app.container.js [hunk 11]
```diff
@@ -26,34 +26,31 @@ class AppContainer extends Component {
   startDownload(id) {
     this.setState({
       progress: 0,
-      showProgressBar: true
+      showProgressBar: true,
+        progressMessage: 'Sending request...',
```
Comment (leerob) [inline app/containers/app.container.js]: Same here

File: app/containers/app.container.js [hunk 12]
```diff
@@ -76,25 +73,22 @@ class AppContainer extends Component {
     clearInterval(this.interval);
 
     // Reset the progress bar to the LinkInput
-    setTimeout(
-      () => this.setState({
+    setTimeout(() => this.setState({
         showProgressBar: false
```
Comment (leerob) [inline app/containers/app.container.js]: I think there's also an extra indentation on `showProgressBar` here. It might be work checking out [ESLint](https://eslint.org/). Your editor/IDE of choice probably has a plugin for it.

File: main.js [hunk 13]
```diff
@@ -44,7 +42,7 @@ function createWindow(){
         ]
     }];
 
-    //if developing add dev menu option to menu bar
+    //If developing add dev menu option to menu bar
```
Comment (leerob) [inline main.js]: Nit: There should still be a space in between `/` and `I` 😄

File: app/containers/app.container.js (hunk 10 above)
Comment (IT-MikeS) [reply app/containers/app.container.js]: Lol I agree, I think it's because my editor is setup for 4 spaces for indent instead of 2

File: app/containers/app.container.js (hunk 11 above)
Comment (IT-MikeS) [reply app/containers/app.container.js]: Yep

File: app/containers/app.container.js (hunk 12 above)
Comment (IT-MikeS) [reply app/containers/app.container.js]: Yeah, I'll change my spacing and resubmit the changes

File: main.js (hunk 13 above)
Comment (IT-MikeS) [reply main.js]: Lol will fix

**Comment by IT-MikeS** [comment]
Okay that should fix the style issues and setup others to meet the same requirements.

**Comment by IT-MikeS** [comment]
Submitted PR #5 will leave this one open in the event that my ideas and direction is not desirable.
